ksqldb-rest-app/src/main/java/io/confluent/ksql/api/impl/InsertsSubscriber.java

```diff
@@ -107,15 +115,19 @@ public static InsertsSubscriber createInsertsSubscriber(final ServiceContext ser
         valueSerde.serializer(), acksPublisher, workerExecutor);
   }
 
-  private InsertsSubscriber(final Context context,
-      final Producer<byte[], byte[]> producer, final DataSource dataSource,
+  private InsertsSubscriber(
+      final Context context,
+      final Producer<byte[], byte[]> producer,
+      final DataSource dataSource,
       final Serializer<Struct> keySerializer,
       final Serializer<GenericRow> valueSerializer,
       final BufferedPublisher<InsertResult> acksPublisher,
-      final WorkerExecutor workerExecutor) {
+      final WorkerExecutor workerExecutor
+  ) {
     super(context);
     this.producer = Objects.requireNonNull(producer);
     this.dataSource = Objects.requireNonNull(dataSource);
+    this.keySchema = ConnectSchemas.columnsToConnectSchema(dataSource.getSchema().key());
```

Comment:
Fixes issue flagged in previous PR review: https://github.com/confluentinc/ksql/pull/6200#discussion_r489048697